java-client/barrage-dagger/src/main/java/io/deephaven/client/impl/BarrageSubcomponent.java

```diff
@@ -33,6 +32,9 @@ interface Builder extends BarrageSessionFactoryBuilder {
 
         Builder allocator(@BindsInstance BufferAllocator bufferAllocator);
 
+        Builder authenticationTypeAndValue(
+                @BindsInstance @Nullable @Named("authenticationTypeAndValue") String authenticationTypeAndValue);
```

Comment:
Is this sufficient? Does this allow for bi-directional authentication schemes?

Reply:
No.

I think we want a working example we can interact with to come up with a better interface. Either way, the structure in main right now only supports anonymous authentication, and this adds on the ability for one-and-done type auth.

That said, I'm not generally happy w/ the structure of BarrageSubcomponent, FlightSubcomponent, SessionSubcomponent (I know I was the one who made them). We may want to remove some of these layers of abstractions or provide better interfaces for users to hook into.